File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/data/entity/impl/Weapon.java
package com.dkellycollins.triggerfinger.data.entity.impl;

import com.dkellycollins.triggerfinger.data.entity.IWeapon;

import java.io.Serializable;

public class Weapon implements IWeapon, Serializable {
    private final int _id;
    private final int _collidableId;
    private final int _timerId;

    public Weapon(int id, int collidableId, int timerId) {
        _id = id;
        _collidableId = collidableId;
        _timerId = timerId;
    }

    @Override
    public int getCollidableId() {
        return _collidableId;
    }

    @Override
    public int getTimerId() {
        return _timerId;
    }

    @Override
    public int getId() {
        return _id;
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/data/entity/impl/Weapon.kt
package com.dkellycollins.triggerfinger.data.entity.impl

import com.dkellycollins.triggerfinger.data.entity.IWeapon

import java.io.Serializable

data class Weapon(override val id: Int, override val collidableId: Int, override val timerId: Int) : IWeapon, Serializable
